### DISCUSSION TITLE
Session key decryption failed error

### DISCUSSION THREAD

**Discussion by wo-jaydeep** [openpgpjs/openpgpjs#1652] [Q&A]
- OpenPGP.js version: 5.9.0
- Affected platform (Browser or Node.js version): 12.16.1

I am unable to decrypt the file and I am getting the error "Error decrypting message: Session key decryption failed".

I even checked the past discussion but none of them is working.
https://github.com/openpgpjs/openpgpjs/issues/1126
https://github.com/openpgpjs/openpgpjs/discussions/1352
https://github.com/openpgpjs/openpgpjs/discussions/1520

```
const openpgp =  require('openpgp');
const { readFileSync } = require('fs');

const decrypt = async () => {
    try {
        const passphrase = 'Admin@123';
        const privateKeyArmored = readFileSync('./files/private-key.asc', 'utf8');
        const privateKey = await openpgp.decryptKey({
            privateKey: await openpgp.readPrivateKey({ armoredKey: privateKeyArmored }),
            passphrase
        });

        const publicKeyArmored = readFileSync('./files/public-key.asc', 'utf8');
        const publicKey = await openpgp.readKey({ armoredKey: publicKeyArmored });

        const encryptedAndSignedMessage = readFileSync('./files/dummy.txt.pgp', 'utf8');
        const message = await openpgp.readMessage({
            armoredMessage: encryptedAndSignedMessage,
            config: {
                ignoreUnsupportedPackets: true,
                ignoreMalformedPackets: true
            }
        });

        const { data } = await openpgp.decrypt({
            message,
            config: {
                allowInsecureDecryptionWithSigningKeys: true,
                allowUnauthenticatedMessages: true 
            },
            verificationKeys: publicKey,
            decryptionKeys: privateKey,
        });

        console.log("data ", data);
    } catch (e) {
        console.log('error', e);
    }
}

decrypt();
```

If you need the files do let me know I will share them.

**Comment by twiss** [comment]
Hey :wave: Yes, if you could share example files for which this happens, that would be helpful.

And, is this a regression? Did it work in a previous version, and if so which one?

**Comment by wo-jaydeep** [comment]
> Hey 👋 Yes, if you could share example files for which this happens, that would be helpful.
> 
> And, is this a regression? Did it work in a previous version, and if so which one?

@twiss 
Below is the link to the files:
Message - https://drive.google.com/file/d/1cSDcAB0tB7Z-5KZ-_xInN9luzYprqoJU/view?usp=sharing
Private Key - https://drive.google.com/file/d/1Qqt0ve1cp1o9mbILdnnzQGbXrR2s0l9S/view?usp=sharing
Public Key - https://drive.google.com/file/d/1HzFXdp7HVsuqti0YWBbE3EBT2BQfF1uw/view?usp=sharing

Yes, I tried out with the older versions 4.10.8 & 4.10.4 but I was getting the same error.

**Comment by twiss** [comment]
Hey :wave: The message is encrypted using a (sub)key with key ID `D6F8218B5147C9B2`, which does not appear in the key you sent, so it was encrypted with a different key.
